pallets/ocex/src/tests.rs

```diff
@@ -322,6 +322,102 @@ fn test_register_trading_pair_trading_pair_already_registered() {
 	});
 }
 
+#[test]
+fn test_update_trading_pair() {
+	new_test_ext().execute_with(|| {
+		assert_ok!(OCEX::register_trading_pair(
+			Origin::root(),
+			AssetId::asset(10),
+			AssetId::asset(20),
+			1_u128.into(),
+			100_u128.into(),
+			1_u128.into(),
+			100_u128.into(),
+			100_u128.into(),
+			10_u128.into()
+		));
+
+		assert_ok!(OCEX::update_trading_pair(
+			Origin::root(),
+			AssetId::asset(10),
+			AssetId::asset(20),
+			1_u128.into(),
+			100_u128.into(),
+			1_u128.into(),
+			100_u128.into(),
+			100_u128.into(),
+			10_u128.into()
+		));
+
+		assert_last_event::<Test>(
+			crate::Event::TradingPairUpdated {
+				base: AssetId::asset(10),
+				quote: AssetId::asset(20),
+			}
+			.into(),
+		);
+		let trading_pair =
+			TradingPairs::<Test>::get(AssetId::asset(10), AssetId::asset(20)).unwrap();
+		let event: IngressMessages<AccountId32> = IngressMessages::UpdateTradingPair(trading_pair);
+		assert_eq!(OCEX::ingress_messages()[1], event);
+	});
+}
+
+#[test]
+fn test_update_trading_pair_trading_pair_not_registered() {
+	new_test_ext().execute_with(|| {
+		assert_noop!(
+			OCEX::update_trading_pair(
+				Origin::root(),
+				AssetId::asset(10),
+				AssetId::asset(20),
+				1_u128.into(),
+				100_u128.into(),
+				1_u128.into(),
+				100_u128.into(),
+				100_u128.into(),
+				10_u128.into()
+			),
+			Error::<Test>::TradingPairNotRegistered
+		);
+	});
+}
+
+#[test]
+fn test_update_trading_pair_bad_origin() {
+	let account_id = create_account_id();
+	new_test_ext().execute_with(|| {
+		assert_noop!(
+			OCEX::update_trading_pair(
+				Origin::none(),
+				AssetId::asset(10),
+				AssetId::asset(20),
+				1_u128.into(),
+				100_u128.into(),
+				1_u128.into(),
+				100_u128.into(),
+				100_u128.into(),
+				10_u128.into()
+			),
+			BadOrigin
+		);
+		assert_noop!(
+			OCEX::update_trading_pair(
+				Origin::signed(account_id.into()),
+				AssetId::asset(10),
+				AssetId::asset(20),
+				1_u128.into(),
+				100_u128.into(),
+				1_u128.into(),
+				100_u128.into(),
+				100_u128.into(),
+				10_u128.into()
+			),
+			BadOrigin
+		);
+	});
+}
+
```

Comment:
I think should be added tests to test added in this PR assertions (`ensures` e.g. [this](https://github.com/Polkadex-Substrate/Polkadex/blob/5cf79b53346c41deb061c7c27a4f8e28fc6d9da0/pallets/ocex/src/lib.rs#L441-L444)). Bassically all places reported by `codecov`.

Reply:
Updated ✅